src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -139,10 +145,43 @@ private void InitializeCore()
         }
 
         private void UpdateStateMachine()
+        {
+            if (!TryUpdateStateMachine())
+            {
+                Task.Factory.SafeStartNew(LoopOnBackgroundAsync, CancellationToken.None, TaskScheduler.Default);
+            }
+        }
+
+        private async Task LoopOnBackgroundAsync()
+        {
+            for (var i = 0; i < 10; i++)
+            {
+                await Task.Delay(TimeSpan.FromSeconds(2));
+
+                // periodically look for the Resume button
+                await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+                if (TryUpdateStateMachine())
+                {
+                    return;
+                }
+            }
```

Comment:
so, basically, we try 10 times, and if we still can't figure out, we don't want to show anything to users. that's it? right?, then, can we just return if 10 tries failed?